### PR TITLE
[crypto] Remove unused modules

### PR DESCRIPTION
**Pull request by davidiw**
As part of the initiative to harden the crypto library, the first line item is to remove all unused modules: https://github.com/libra/libra/issues/2683

This does that. Note: there is more work that can be done such as eliminating sha2 and (maybe) digest, but we need to tweak either the hkdf or extend tiny-keccak appropriately.

### CODE REVIEW

**Comment by davidiw** [comment]
Perhaps before I commit, I can leave a comment about experimental code. I'm not keen on maintaining experimental code. We have a git repo so we can always resurrect anything here if we need it.

**Comment by kchalkias** [comment]
Yes please add a comment here. I understand we can always find it, just saying that sometimes deleted unused features/algorithms that might be required in the future are forgotten and people build it from scratch if this is not documented properly.

**Comment by davidiw** [comment]
> Yes please add a comment here. I understand we can always find it, just saying that sometimes deleted unused features/algorithms that might be required in the future are forgotten and people build it from scratch if this is not documented properly.

Very fair! I would really hate for us to remove it in this PR and then rather than bring back these versions we write it from scratch. Especially considering that these implementations are rather clean. Stack updated with a pointer.

File: crypto/crypto/README.md [hunk 1]
```diff
@@ -32,3 +32,5 @@ Libra makes use of several cryptographic algorithms:
     ├── traits.rs           # New API design and the necessary abstractions
     └── unit_tests/         # Tests
 ```
+
+Note: This crate historically had support for BLS12381, ECVRF, and SlIP-0010, though were removed due to lack of use. The last git revision before there removal is 00301524.
```
Comment (kchalkias) [inline crypto/crypto/README.md]: Thanks!

**Pull request closed without merging**